eos-updater/eos-updater-autocleanup.service.in

```diff
@@ -0,0 +1,37 @@
+[Unit]
+Description=Automatically cleanup after staged Endless OS Updater deployment
+Documentation=man:ostree-admin-cleanup(1) man:eos-updater(8)
+
+# Run ostree admin cleanup only if /sysroot/.cleanup is present and then
+# delete it when cleanup is successful.
+#
+# FIXME: Drop this when https://github.com/ostreedev/ostree/issues/2510
+# is resolved.
+ConditionPathExists=/sysroot/.cleanup
+
+# We want this to be triggered by multi-user.target but not block it via
+# the default After added to target units since pruning the repo can be
+# slow. See the Default Dependencies sections in systemd.service(5) and
+# systemd.target(5).
+DefaultDependencies=no
+Requires=sysinit.target
+After=sysinit.target basic.target
+Conflicts=shutdown.target
+Before=shutdown.target
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+ExecStart=@ostree@ admin cleanup
+ExecStart=/bin/rm -f /sysroot/.cleanup
+
+# Only /sysroot and /boot need to be written to.
+ProtectSystem=strict
+ReadWritePaths=/sysroot /boot
```

Comment:
But if you can write to `/sysroot`, what *can't* you write to on the main disk?

Reply:
True. I guess the only way to really narrow this is to make it it truly pruning only and then you could limit it to `/sysroot/repo`. `ostree admin cleanup` does do other things in `/sysroot/boot`, `/sysroot/ostree/boot.*` and `/sysroot/ostree/deploy`.

Reply:
While it's true that everything is really under `/sysroot`, I don't believe you meant this as a blocker. It does mean that inadvertent writing to `/etc` isn't possible. Unless you have objections, I think we should just carry on with this.

Reply:
Agreed